test/Riok.Mapperly.Tests/Mapping/InstantiableMapperWithStaticMethodsTest.cs

```diff
@@ -0,0 +1,80 @@
+using Riok.Mapperly.Diagnostics;
+
+namespace Riok.Mapperly.Tests.Mapping;
+
+[UsesVerify]
+public class InstantiableMapperWithStaticMethodsTest
+{
+    [Fact]
+    public Task StaticPartialMethod()
+    {
+        var source = TestSourceBuilder.CSharp(
+            """
+            using Riok.Mapperly.Abstractions;
+
+            record A(int Value);
```

Comment:
Can you add a test with a nested object type?

Reply:
I think you misunderstood me here, my bad for not being enough specific...
What I actually meant is a test with a sth. like this:
Source: `record A(B Value); record B(int Value);`
Target: `record C(D Value); record D(int Value);`

Related to what I was trying to describe [here](https://github.com/riok/mapperly/pull/681#discussion_r1384755733).
A second test case could be like this (to address https://github.com/riok/mapperly/pull/681#discussion_r1384755733): 
Source: `record A(B Value); record B(C Value); record C(int Value)`
Target: `record D(E Value); record E(F Value); record F(int Value)`
With a user implemented mapping from C to F.

Reply:
Thank you for explaining. Both test cases added